packages/langium/src/documents/document-builder.ts

```diff
@@ -116,6 +151,7 @@ export class DefaultDocumentBuilder implements DocumentBuilder {
 
     protected async buildDocuments(documents: LangiumDocument[], cancelToken: CancellationToken): Promise<void> {
         await this.indexManager.update(documents.filter(e => e.state < DocumentState.Indexed), cancelToken);
+        await this.notifyBuildPhase(documents, DocumentState.Indexed, cancelToken);
```

Comment:
Should we pass the filtered documents here, instead of all documents? Some of the documents have been indexed previously, but will still be passed to the callback. The same applies to the call of `notifyBuildPhase` of `runCancelable`.